### ISSUE TITLE
infinite loop in BuildMonitor.wait! because build.lock never gets cleaned up

### ISSUE THREAD

**Issue by rusanu** [thoughtbot/ember-cli-rails#377]
I'm trying to serve an ember app from rails. The build stage succeeds, but the wait for build never completes because it waits for `<Rails.root>/<ember-app>/tmp/build.lock` to be cleaned up, which never happens (infinite loop in `BuildMonitor.wait!`). As far as I can tell, the ember build never touches this files and does not removes it. I placed a root owned build.lock in `<ember-app>/tmp` and ember build completed successfully as non-root, never issuing a permission error (by contrast ember-cli-rails fails if the root owned file exists with access error in `Shell.lock_buildfile`). 

I searched through Broccoli and ember-cli repos (builder task, builder model) but I did not find any reference to ember-cli build having to cleanup a `<ember-app>/tmp/build.lock` 

I did not find any recent change, and I have similar setup working fine. I'm a bit at a loss on how this is supposed to work, basically who is responsible for removing this file and why doesn't it get removed. As far as I can tell Brocolli builder cleans up `<ember-app>/tmp`, but only of the files it created itself.

  ember -v
  version: 1.13.13
  node: 4.2.4
  npm: 2.14.10
  os: linux x64

**Comment by seanpdoyle** [comment]
The addon is responsible for cleaning up the lockfile [when the `outputReady` event fires](https://github.com/rondale-sc/ember-cli-rails-addon/blob/03c722edfce4dd856cf467d2c804bdde591ad6ec/index.js#L45-L51).

If the file is never removed, the build is likely not successful. When you experience the infinite loop, does `<ember-app>/tmp/error.txt` exist?

**Comment by rusanu** [comment]
The build succeeds. If I remove the file manually, the loop stops, the call succeeds and the ember app is rendered. 

I added the ember-cli-rails-addon in the `<ember-app>/package.json` and everything works fine now. I think the problem originated because I create the ember app from ember-cli directly. I see now there is a [rake ember:install](https://github.com/thoughtbot/ember-cli-rails/blob/master/lib/tasks/ember-cli.rake) task that I skipped.

Thanks!
